### PR TITLE
Uses fzf fuzzy search (via iterfzf, which ships fzf inside) for choice screens

### PR DESCRIPTION
**Pull request by dbalan**
1. Project
 2. Service type

### CODE REVIEW

File: mite_cli.py [hunk 1]
```diff
@@ -50,40 +51,42 @@ def editor(txt=""):
...
 
 
 def get_service(mite):
     services = mite.list_services()
     print("Choose a service to add the entry to:")
-    for idx, thng in enumerate(services):
-        print("[{}] {}".format(idx+1, thng["service"]["name"]))
-    return choose_from_list(services, "service")
+
+    sv_choices = {}
```
Comment (hellerve) [inline mite_cli.py]: At this point, couldn’t we do

```python
sv_choices = {e["name"]:  e["id"] for s in services for e in s["service"]}
```

File: mite_cli.py [hunk 2]
```diff
@@ -50,40 +51,42 @@ def editor(txt=""):
...
-        customer_name = ""
-        cs = [c for c in customers
-              if c["customer"]["id"] == project["customer_id"]]
-        if cs:
-            customer_name = cs[0]["customer"]["name"]
-        print("[{}] {} ({})".format(idx+1, project["name"], customer_name))
 
-    return choose_from_list(projects, "project")
+    # build {cust id -> name}
+    cust = {}
```
Comment (hellerve) [inline mite_cli.py]: Here as well:

```python
cust = {e["id"]:  e["name"] for s in services for e in s["customer"]}
```

**Review by hellerve — CHANGES_REQUESTED** [review]
Thanks for working on this!

In order for this to work with `setup.py`, the requirement needs to be added there as well.

I also added a few minor comments, but they are not urgent/required.

File: mite_cli.py (hunk 2 above)
Comment (dbalan) [reply mite_cli.py]: This one is definitely succinct. I'll change the patch.

**Comment by dbalan** [comment]
@hellerve I've updated the PR.

File: mite_cli.py [hunk 3]
```diff
@@ -50,44 +51,43 @@ def editor(txt=""):
...
+        full_name = "{} ({})".format(p["name"], cust_name)
+        choices[full_name] = p["id"]
+
+    print("Choose a project to add the entry to:")
+    return choose_with_fzf(choices)
 
 
 def get_service(mite):
     services = mite.list_services()
+    sv_choices = { s["service"]["name"] : s["service"]["id"]
```
Comment (hellerve) [inline mite_cli.py]: Nitpick: why is there a leading space in the comprehension here and not above?

**Review by hellerve — APPROVED** [review]
Looks great to me now!

**Comment by dbalan** [comment]
okay, fixed a bunch of style errors, including the nitpick :-) Can I go ahead and merge?

**Comment by hellerve** [comment]
Did that for you, sorry!

Thanks for your work on this! 😸

**Comment by dbalan** [comment]
thanks! Also no worries, I love a thorough code review :-)

**Pull request merged**